src/games/supported/DemonAttack.cpp

```diff
@@ -121,3 +129,41 @@ void DemonAttackSettings::loadState(Deserializer & ser) {
   m_lives = ser.getInt();
 }
 
+// returns a list of mode that the game can be played in
+ModeVect DemonAttackSettings::getAvailableModes(){
+    ModeVect modes;
+    modes.push_back(1);
+    modes.push_back(3);
+    modes.push_back(5);
+    modes.push_back(7);
+
+    return modes;
+}
+
+// set the mode of the game
+// the given mode must be one returned by the previous function
+void DemonAttackSettings::setMode(game_mode_t m, System &system, StellaEnvironment& environment){
+    if(m==1 || m==3 || m==5 || m==7){
```

Comment:
Agreed, we loose the ordering of the modes, but it doesn't make much sense in the majority of the game anyways.
If you agree to c++11, we can then have a very nice syntax for the getAvailableModes, using the {} constructor:
`   return {1,3,5,7}`, which looks nicer than a bunch of inserts/push_back.